xtext-website/documentation/303_runtime_concepts.md

```diff
@@ -185,7 +185,9 @@ Feature:
 
 The grammar declares that for the reference *type* only instances of the type *Type* are allowed. However, this simple declaration doesn't say anything about where to find the type. That is the duty of scopes.
 
-An [IScopeProvider]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/scoping/IScopeProvider.java) is responsible for providing an [IScope]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/scoping/IScope.java) for a given context [EObject]({{site.src.emf}}/plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/EObject.java) and [EReference]({{site.src.emf}}/plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/EReference.java). The returned [IScope]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/scoping/IScope.java) should contain all target candidates for the given object and cross-reference.
+An [IScopeProvider]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/scoping/IScopeProvider.java) is responsible for providing an [IScope]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/scoping/IScope.java) for a given context [EObject]({{site.src.emf}}/plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/EObject.java) and [EReference]({{site.src.emf}}/plugins/org.eclipse.emf.ecore/src/org/eclipse/emf/ecore/EReference.java). The "context" EObject will represent the parsed token that is interpreted as a reference in your DSL. The "reference" EReference represents the reference **in the parser rules** from the parent parser rule to that token; this allows you to identify in what context in the grammar the DSL reference is being encountered. The reference parameter does **not** refer to the "reference" that is being scoped in your DSL.  
```

Comment:
It's the "reference" in the sense of the reference that's made in the grammar to the item requiring scoping. (eg, "the name of a variable in an assignment") It's not the "reference" in the sense of the thing in the DSL that refers to something that is or is not in scope (eg, "foo", this being what is entered in that context in the assignment). This caused me a lot of confusion when working from this.